docs/api/standard_library_types.md

```diff
@@ -386,6 +386,412 @@ except ValidationError as e:
     """
 ```
 
+## Set
+
+### `set`
+
+Allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a `set`.
+When a generic parameter is provided, the appropriate validation is applied to all items of the set.
+
+### `typing.Set`
+
+Handled the same as `set` above.
+
+```py
+from typing import Optional, Set
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    simple_set: Optional[set] = None
+    set_of_ints: Optional[Set[int]] = None
+
+
+print(Model(simple_set={'1', '2', '3'}).simple_set)
+#> {'1', '2', '3'}
+print(Model(simple_set=['1', '2', '3']).simple_set)
+#> {'1', '2', '3'}
+print(Model(set_of_ints=['1', '2', '3']).set_of_ints)
+#> {1, 2, 3}
+```
+
+## Frozenset
+
+### `frozenset`
+
+Allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a `frozenset`.
+When a generic parameter is provided, the appropriate validation is applied to all items of the frozen set.
+
+### `typing.FrozenSet`
+Handled the same as `frozenset` above.
+
+```py
+from typing import FrozenSet, Optional
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    simple_frozenset: Optional[frozenset] = None
+    frozenset_of_ints: Optional[FrozenSet[int]] = None
+
+
+m1 = Model(simple_frozenset=['1', '2', '3'])
+print(type(m1.simple_frozenset))
+#> <class 'frozenset'>
+print(sorted(m1.simple_frozenset))
+#> ['1', '2', '3']
+
+m2 = Model(frozenset_of_ints=['1', '2', '3'])
+print(type(m2.frozenset_of_ints))
+#> <class 'frozenset'>
+print(sorted(m2.frozenset_of_ints))
+#> [1, 2, 3]
+```
+
+
+## Support for iterable types
+
+### `typing.Sequence`
+This is intended for use when the provided value should meet the requirements of the `Sequence` protocol, and it is
+desirable to do eager validation of the values in the container. Note that when validation must be performed on the
+values of the container, the type of the container may not be preserved since validation may end up replacing values.
+We guarantee that the validated value will be a valid `typing.Sequence`, but it may have a different type than was
+provided (generally, it will become a `list`).
```

Comment:
```suggestion
### `typing.Sequence`

This is intended for use when the provided value should meet the requirements of the `Sequence` protocol, and it is
desirable to do eager validation of the values in the container. Note that when validation must be performed on the
values of the container, the type of the container may not be preserved since validation may end up replacing values.
We guarantee that the validated value will be a valid `typing.Sequence`, but it may have a different type than was
provided (generally, it will become a `list`).
```